File: programmers/level2/pro42842.py
# 프로그래머스 카펫

def solution(brown, red):
    answer = []
    temp = brown + red # 전체 카펫의 수
    #가로길이는 세로 길이와 같거나, 세로길이보다 길다.
    garo = temp
    sero = 1
    
    while(garo >= sero):
        if((garo - 2 )* (sero -2) == red):
            answer = [garo, sero] # 제일 끝애 두개 뺀거를 곱했을 때 red랑 같아야함.
            break
        garo -=1
        sero = temp // garo
    return answer

File: javascript/pro42842.js
// 프로그래머스 level2 카펫

function solution(brown, red) {
    const answer = [];
    let temp = brown + red
    // 가로길이는 세로 길이와 같거나, 세로길이보다 길다.
    let height = 3;
    let width = parseInt(temp / height)

    while (width >= height) {
        if ((width - 2) * (height - 2) == red) {
            answer.push(width);
            answer.push(height);
            break;
        }
        width -= 1
        height = parseInt(temp / width)
    }
    return answer;
}


console.log(parseInt(10/3))